src/abilities/Snow-Bunny.js

```diff
@@ -13,24 +13,31 @@ export default (G) => {
 		// 	First Ability: Bunny Hop
 		{
 			//	Type : Can be "onQuery", "onStartPhase", "onDamage"
-			trigger: 'onOtherCreatureMove',
+			trigger: 'onCreatureMove onOtherCreatureMove',
 
-			require: function (fromHex) {
+			require: function () {
 				if (!this.testRequirements()) {
 					return false;
 				}
 
-				// If enough uses, jump away when an enemy has entered our trigger area, and
-				// we have a space to jump back to
-				return (
+				const enemyInFront = this._detectEnemyInFront();
+
+				/* If enough uses, jump away when an enemy has entered our trigger area, 
+				and we have a space to jump back to. */
+				const abilityCanTrigger =
 					this.timesUsedThisTurn < this._getUsesPerTurn() &&
-					fromHex &&
-					fromHex.creature &&
-					isTeam(fromHex.creature, this.creature, Team.enemy) &&
-					this.game.activeCreature != this.creature &&
-					this._getTriggerHexId(fromHex) >= 0 &&
-					this._getHopHex(fromHex) !== undefined
-				);
+					// Bunny cannot use this ability if affected by these states.
+					!(this.creature.materializationSickness || this.creature.stats.frozen) &&
+					// This ability only triggers on other creature's turns, it's purely defensive.
+					this.creature !== this.game.activeCreature &&
+					enemyInFront &&
+					/* Only the active enemy creature should trigger the Hop, not other enemy 
+					creatures that happen to be nearby. */
+					enemyInFront === this.game.activeCreature &&
```

Comment:
Not really. Basically a friend or foe could move an enemy unit in front of Snow Bunny, which should trigger the passive.
Scavenger can use Escort Service ability to move units around.

Reply:
MR updated, see my comment below: https://github.com/FreezingMoon/AncientBeast/pull/1939#issuecomment-989197205